aplib.net-demo/Assets/Testing/AplibTests/RagePotionAplibTest.cs

```diff
@@ -0,0 +1,162 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+
+namespace Testing.AplibTests
+{
+    public class RagePotionBeliefSet : BeliefSet
+    {
+        /// <summary>
+        /// The player object in the scene.
+        /// </summary>
+        public Belief<GameObject, GameObject> Player = new(GameObject.Find("Player"), x => x);
+
+        /// <summary>
+        /// The rage potion in the scene.
+        /// </summary>
+        public Belief<GameObject, RagePotion> RagePotion = new(GameObject.Find("RagePotion"), x => x != null ? x.GetComponent<RagePotion>() : null);
+
+        /// <summary>
+        /// The target dummy health in the scene.
+        /// </summary>
+        // public Belief<GameObject, GameObject> TargetDummy = new(GameObject.Find("Target Dummy"), x => x);
+        public MemoryBelief<GameObject, int> TargetDummyHealth = new(GameObject.Find("Target Dummy"), x => x.GetComponentInChildren<HealthComponent>().Health, 60);
+
+
+        /// <summary>
+        /// The inventory in the scene.
+        /// </summary>
+        public Belief<GameObject, Inventory> InventoryObject = new(GameObject.Find("InventoryObject"), x => x.GetComponent<Inventory>());
+
+        /// <summary>
+        /// The player weapon in the scene.
+        /// </summary>
+        public Belief<GameObject, Weapon> PlayerWeapon = new(GameObject.Find("Player"), x => x.GetComponentInChildren<EquipmentInventory>().CurrentEquipment as Weapon);
+
+        /// <summary>
+        /// The target position that the player needs to move towards.
+        /// Find the first Rage Potion in the scene.
+        /// </summary>
+        public Belief<GameObject, Vector3> PotionPosition = new(GameObject.Find("RagePotion"), x => x == null ? Vector3.zero : x.transform.position);
+
+        public Belief<GameObject, Vector3> EnemyPosition = new(GameObject.Find("Target Dummy"), x => x == null ? Vector3.zero : x.transform.position);
+
+        /// <summary>
+        /// The rage potion in the scene.
+        /// </summary>
+        public Belief<GameObject, bool> PotionExists = new(GameObject.Find("RagePotion"), x => x != null && x.activeSelf && x.activeInHierarchy);
+    }
+
+    public class RagePotionAblibTest
+    {
+        private int _increasedDamage;
+        private int _startDamage;
+        private int _enemyStartHealth;
+
+        [SetUp]
+        public void SetUp()
+        {
+            Debug.Log("Starting test RagePotionAplibTest");
+            SceneManager.LoadScene("RagePotionTestScene");
+        }
+
+        /// <summary>
+        /// The player will walk up to a rage potion and use it.
+        /// Then check if the rage potion doubles the players damage when damaging an entity.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator PerformRagePotionAplibTest()
+        {
+            InputManager.Instance.enabled = false;
+            var beliefSet = new RagePotionBeliefSet();
```

Comment:
Nit
```suggestion
            RagePotionBeliefSet beliefSet = new();
```

Reply:
Also please rename this to something like `ragePotionBeliefSet` to prevent name shadowing later on.